bindings/cs/rl.net/ApiStatus.cs

```diff
@@ -52,5 +59,58 @@ public string ErrorMessage
                 return result;
             }
         }
+
+        internal static void Update(ApiStatus status, int errorCode, string message)
+        {
+            unsafe 
+            { 
+                fixed (byte* messageBytes = NativeMethods.StringEncoding.GetBytes(message))
+                {
+                    IntPtr messagePtr = new IntPtr(messageBytes);
+                    UpdateApiStatusSafe(status.ToNativeHandleOrNullptrDangerous(), errorCode, messagePtr);
```

Comment:
This should not be a leak because `messagePtr` is only valid for the lifetime of the `fixed` block. This code relies on the understanding that the `api_status` object copies the string on input. Broadly speaking, the RLClientLib bindings assume that the native-side will introduce a copy if necessary, but strings coming in only need to live until the NativeMethod returns.

Reply:
You're right, GetBytes allocates a `byte[]` and not malloc something.